diagnostic.js

```diff
@@ -30,25 +30,53 @@ mongoose.connect('mongodb://localhost/mongoose-crud');
 
 /// ADD YOUR CODE BELOW
 
-const create = (name, description, startYear, endYear) => {};
+const create = (name, description, startYear, endYear) => {
+  Movement.create({
+    name: name,
+    description: description,
+    startYear: startYear,
+    endYear: endYear
+  }).then(movement => {
+    console.log(movement.toJSON())
+  }).catch(console.error);
+};
+
+
 // Success -> Print new Movement as JSON
 // Failure -> Console.error
 
-const index = () => {};
+const index = () => {
+  Movement.find()
+  .then(movements => {
+    console.log(movements.toJSON())
+  }).catch(console.error);
+};
 // Success -> Print all Movements as JSON
 // Failure -> Console.error
 
-const show = (id) => {};
+const show = (id) => {
+  Movement.findById(id)
+  .then(movement => {
+    console.log(movement.toJSON())
```

Comment:
Missing one edge case - if no document is found with the given id, `findById` will return `null` _which is not the same thing as hitting an error_. You need to handle that case as well; if `movement` is `null`, print out "Not Found" and exit, and if it isn't, print JSON.